crates/turbopack-ecmascript/src/references/cjs.rs

```diff
@@ -278,27 +263,33 @@ impl CodeGenerateable for CjsRequireResolveAssetReference {
         let mut visitors = Vec::new();
 
         let path = &self.path.await?;
-        if let PatternMapping::Invalid = &*pm {
-            let request_string = self.request.to_string().await?;
-            visitors.push(create_visitor!(path, visit_mut_expr(expr: &mut Expr) {
-                // In Node.js, a require.resolve call that cannot be resolved will throw an error.
-                *expr = throw_module_not_found_expr(&request_string);
-            }));
-        } else {
-            // Inline the result of the `require.resolve` call as a string literal.
-            visitors.push(create_visitor!(path, visit_mut_expr(expr: &mut Expr) {
-                if let Expr::Call(call_expr) = expr {
-                    let args = std::mem::take(&mut call_expr.args);
-                    *expr = match args.into_iter().next() {
-                        Some(ExprOrSpread { expr, spread: None }) => pm.apply(*expr),
-                        _ => pm.create(),
-                    };
-                }
-                // CjsRequireResolveAssetReference will only be used for Expr::Call.
-                // Due to eventual consistency the path might match something else,
-                // but we can ignore that as it will be recomputed anyway.
-            }));
-        }
+        // Inline the result of the `require.resolve` call as a literal.
+        visitors.push(create_visitor!(path, visit_mut_expr(expr: &mut Expr) {
+            if let Expr::Call(call_expr) = expr {
+                let args = std::mem::take(&mut call_expr.args);
+                *expr = match args.into_iter().next() {
+                    Some(ExprOrSpread { expr, spread: None }) => pm.create_require(*expr),
```

Comment:
Unfortunately I just noticed this now, but this looks very wrong, this should insert a string, not a require call